examples/most-redis-docker/README.md

```diff
@@ -0,0 +1,132 @@
+
+## Install
+
+### Pre-requisites
+
+1. Install [git](https://help.github.com/articles/set-up-git/)
+2. Clone this repo:
+
+```sh
+git clone git@github.com:mostjs/core.git
+cd examples/most-redis-docker
+```
+
+### Assumptions
+
+The remainder of this document assumes your working directory is
+`most-redis-docker`.  You may have to modify some of the following directions
+if you chose another folder name.
+
+This project uses environment variables.  If you are using the windows command
+prompt, you may have to change some commands slightly to use env vars. For
+example, rather than type
+`REDIS_MYCHANNEL=most-redis-docker-example npm run consume`, you would type
+`set REDIS_MYCHANNEL=most-redis-docker-example&&npm run consume`.
+
+### Option 1: Docker
+
+1.  Install [docker](https://www.docker.com/get-started).  (You will need to
+    create a [DockerHub](https://hub.docker.com) account.)
+2.  Build the docker image for this project:
+
+```sh
+docker build -t most-redis-docker .
+```
+
+### Option 2: node.js
+
+1. Install [node 10.x](https://nodejs.org/en/) or the latest LTS version.
+2. Install the dependencies for this project:
+
+```sh
+npm install
+```
+
+3. Install [redis](https://redis.io/topics/quickstart).
+
+If you already have an instance of redis running, you may edit the docker-
+compose.yml file in the `most-redis-docker` folder.  Change any values for
+`REDIS_HOST` and `REDIS_PORT` according to your redis instance
+
+## Run
+
+When running this example, the publish service periodically sends JSON messages
+to redis, and the consume services receive the JSON messages and transform
+them.  In your terminal(s), you should expect to see informational logs from
+each service as they start up and as they process the JSON messages.
+
+### Option 1: Docker
+
+1.  From the `most-redis-docker` folder, create and run the docker-compose
+    network defined in `docker-compose.yml`:
+
+```sh
+docker-compose up
+```
+
+Use `Ctrl-C` to stop the docker-compose network.
+
+You may also use `docker-compose up -d` and `docker-compose down` to start and
+stop the docker-compose network in the background.
+
+### Option 2: node.js
+
+1.  Start redis.  See the [redis docs](https://redis.io/topics/quickstart) for
+    troubleshooting and options.
+
+```sh
+redis-server
+```
+
+2. From the `most-redis-docker` folder:
+
+```sh
+REDIS_MYCHANNEL=most-redis-docker-example node consume
+```
+
+3. Open a new terminal or shell, and from the `most-redis-docker` folder type:
+
+```sh
+REDIS_MYCHANNEL=most-redis-docker-example node consume
+```
+
+4. Open a third terminal or shell, and from the `most-redis-docker` folder type:
+
+```sh
+REDIS_MYCHANNEL=most-redis-docker-example node publish
+```
+
+## Experiment!
+
+### Env vars
+
+Both the `consume` and `publish` services accept env vars.  If you are using
+Docker, these may be modified in the `docker-compose.yml` file.  The following
+env vars are used:
+
+- `REDIS_CHANNEL`: string, _required_
```

Comment:
REDIS_CHANNEL is now at the top to stand out better.

Reply:
I guess technically speaking, REDIS_CHANNEL isn't required now.  Would you prefer to leave "_required_" here to encourage people to set it, or replace it with "default is ...."?

Reply:
Fixing it...